packages/suite/src/utils/wallet/tokenUtils.ts

```diff
@@ -83,16 +109,23 @@ export const getTokens = (
     const unverifiedWithBalance: EnhancedTokenInfo[] = [];
     const unverifiedWithoutBalance: EnhancedTokenInfo[] = [];
 
-    tokensWithoutNFTs.forEach(token => {
-        const isKnown = isTokenDefinitionKnown(coinDefinitions?.data, symbol, token.contract);
-        const isHidden = coinDefinitions?.hide.includes(token.contract);
-        const isShown = coinDefinitions?.show.includes(token.contract);
+    filteredTokens.forEach(token => {
+        const isKnown = isTokenDefinitionKnown(tokenDefinitions?.data, symbol, token.contract);
+        const isHidden = tokenDefinitions?.hide.includes(token.contract);
+        const isShown = tokenDefinitions?.show.includes(token.contract);
 
         const query = searchQuery ? searchQuery.trim().toLowerCase() : '';
 
-        if (searchQuery && !isTokenMatchesSearch(token, query)) return;
+        if (
+            searchQuery &&
+            (!isTokenMatchesSearch(token, query) ||
+                (isNft && isNftMatchesSearch(token as TokenInfo, query)))
```

Comment:
This is a bit ugly and actually doesn't work, I would rewrite it like this:
```suggestion
            !(isNft ? isNftMatchesSearch(token, query) : isTokenMatchesSearch(token, query))
```

Reply:
Thank you, implemented